### PR TITLE
Setup react-frontend

### PR DESCRIPTION
**Pull request by Ibtisam34**
- [ ] This exercise is going to have I  created a React front-end with a Rails back-end and connect them to display a random message.

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const messageSlice = createSlice({
+  name: 'message',
+  initialState: { message: '' },
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchMessage.fulfilled,
+      (state, action) => ({ ...state, message: action.payload }
+      ));
+  },
+});
```
Comment (Gambit142) [inline src/store.js]: - [ ] You didn't create a store for your reducer. you need to create a store that holds the whole state tree of your application. It is through the store that components can connect to the application's state and change the state through the dispatch of actions. 👍

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+);
```
Comment (Gambit142) [inline src/index.js]: - [ ] When I start the application, the screen is blank. After creating the store, you need to connect to the store using the Provider component from redux. Kindly apply this change. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Ibtisam34 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Use of reducers, actions and `createAsyncThunk` ✔️ 
- It has `App` and `Greeting` component ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: README.md [hunk 3]
```diff
@@ -1 +1,168 @@
...
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ (OPTIONAL)](#faq)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 [ Hello-world-front-end ] <a name="about-project"></a>
+
+> Describe your project in 1 or 2 sentences.
```
Comment (andersonlebon) [inline README.md:34]: ## [OPTIONAL]

- [ ] To make your readme file more professional I think is good that all readme template text descriptions are removed

File: README.md [hunk 4]
```diff
@@ -1 +1,168 @@
...
+
+If you like this project hit the star button
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+> Give credit to everyone who inspired your codebase.
```
Comment (andersonlebon) [inline README.md:154]: ## [OPTIONAL]

- [ ] To make your readme file more professional I think is good that all readme template text descriptions are removed

File: README.md [hunk 5]
```diff
@@ -1 +1,168 @@
...
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
+
+_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (andersonlebon) [inline README.md:168]: ## [OPTIONAL]
- [ ] To make your readme file more professional I think is good that all readme template text descriptions are removed

**Review by andersonlebon — APPROVED** [review]
## Hi @Ibtisam34 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**